math/factorial_recursion.c

```diff
@@ -0,0 +1,76 @@
+/*
+    Algorithm: Factorial Calculation using Memoization
+
+    This C program calculates factorials of non-negative integers using a recursive algorithm with memoization.
+    Factorial of a non-negative integer 'n' is denoted as 'n!' and is defined as the product of all positive integers
+    less than or equal to 'n'. For example, 5! = 5 x 4 x 3 x 2 x 1 = 120.
+
+    Author: Sricharan Nibhanupudi
+    
+    Uses:
+    - The program provides a function 'computeFactorial' that calculates the factorial of a given non-negative integer
+      using recursive memoization. It stores previously computed factorials in an array 'factResults' to avoid redundant
+      calculations, which significantly improves performance for large values of 'n'.
+    - The 'testFactorial' function uses assertions to verify the correctness of the 'computeFactorial' function by
+      comparing its results with known factorial values.
+    - The program also demonstrates the use of standardized integer data types like 'uint64_t' from 'stdint.h' for
+      improved code portability and readability.
+
+    References:
+    - The concept of factorial: https://en.wikipedia.org/wiki/Factorial
+    - Memoization: https://www.geeksforgeeks.org/memoization-1d-2d-and-3d/
+*/
+
+#include <stdio.h>

```

Comment:
it would be preferable if you document the headers as well
```c
#include<header.h> /// <reason for using header>

Reply:
I have made all the required changes. Kindly look into it. Thank you.

Reply:
@realstealthninja, can you please review the pull request?